internal/pkg/workspace/workspace.go

```diff
@@ -305,10 +303,17 @@ func (ws *Workspace) write(data []byte, elem ...string) (string, error) {
 	filename := filepath.Join(pathElems...)
 
 	if err := ws.fsUtils.MkdirAll(filepath.Dir(filename), 0755 /* -rwxr-xr-x */); err != nil {
-		return "", fmt.Errorf("failed to create directories for file %s: %w", filename, err)
+		return "", fmt.Errorf("create directories for file %s: %w", filename, err)
+	}
+	exist, err := ws.fsUtils.Exists(filename)
+	if err != nil {
+		return "", fmt.Errorf("check if manifest file exists: %w", err)
+	}
+	if exist {
+		return filename, &ErrFileExists{fileName: filename}
```

Comment:
What do you think about returning like this:
```suggestion
		return "", &ErrFileExists{Filename: filename}
```
This way we can get rid of the hidden assumption that a user needs to check the first return value if the error is `ErrFileExists` which I don't think is a common practice.